pkgs/development/python-modules/pypdfium2/default.nix

```
@@ -0,0 +1,136 @@
+{
+  lib,
+  pkgs,
+  buildPythonPackage,
+  fetchFromGitHub,
+  fetchurl,
+  setuptools-scm,
+  pdfium-binaries,
+  numpy,
+  pillow,
+  pytest,
+  python,
+}:
+
+let
+  pdfiumVersion = "${pdfium-binaries.version}";
+
+  headers = fetchurl {
+    url = "https://pdfium.googlesource.com/pdfium/+archive/refs/heads/chromium/${pdfiumVersion}/public.tar.gz";
+    hash = "sha256-yeuIk2Dp36cVVJX4/3OTq769wSJB0QhuZfU5lDyDotQ=";
+  };
+
+  # They demand their own fork of ctypesgen
+  ctypesgen = buildPythonPackage rec {
+    pname = "ctypesgen";
+    version = "1.1.1+g${src.rev}"; # the most recent tag + git version
+    pyproject = true;
+
+    src = fetchFromGitHub rec {
+      inherit pname;
+      version = rev;
+      owner = "pypdfium2-team";
+      repo = "ctypesgen";
+      rev = "848e9fbb1374f7f58a7ebf5e5da5c33292480b30";
+      hash = "sha256-3JA7cW/xaEj/DxMHEypROwrKGo7EwUEcipRqALTvydw=";
+    };
+
+    build-system = [
+      setuptools-scm
+    ];
+
+    env.SETUPTOOLS_SCM_PRETEND_VERSION = "${version}";
+  };
+
+in
+buildPythonPackage rec {
+  pname = "pypdfium2";
+  version = "4.30.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    inherit pname;
+    owner = "pypdfium2-team";
+    repo = "pypdfium2";
+    tag = version;
+    hash = "sha256-v8f/XruGJYK3H9z4Q1rLg4fEnPHa8tTOlNTBMVxPEgA=";
+  };
+
+  build-system = [
+    ctypesgen
+    setuptools-scm
+  ];
+
+  dependencies = [
+    pdfium-binaries
+  ];
```

```suggestion
  propagatedBuildInputs = [
    pdfium-binaries
  ];
```
`pdfium-binaries` is not a python library.